src/Simulation/CsharpGeneration/RewriteStep.fs

```diff
@@ -0,0 +1,49 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+namespace Microsoft.Quantum.QsCompiler.CsharpGeneration
+
+open System
+open System.Collections.Generic
+open System.IO
+open Microsoft.Quantum.QsCompiler
+open Microsoft.Quantum.QsCompiler.CsharpGeneration
+open Microsoft.Quantum.QsCompiler.ReservedKeywords
+open Microsoft.Quantum.QsCompiler.SyntaxTree
+open Microsoft.Quantum.QsCompiler.Transformations.BasicTransformations
+
+
+type Emitter() =
+
+    let _AssemblyConstants = new Dictionary<string, string>()
+
+    static member internal SupportUnitTests = "CsharpGeneration/TestProject"
+
+    interface IRewriteStep with
+
+        member this.Name = "CsharpGeneration"
+        member this.Priority = -1 // doesn't matter because this rewrite step is the only one in the dll
+        member this.AssemblyConstants = _AssemblyConstants :> IDictionary<string, string> 
+        
+        member this.ImplementsPreconditionVerification = false
+        member this.ImplementsPostconditionVerification = false
+        member this.ImplementsTransformation = true
+
+        member this.PreconditionVerification _ = NotImplementedException() |> raise
+        member this.PostconditionVerification _ = NotImplementedException() |> raise
+        
+        member this.Transformation (compilation, transformed) = 
+            let step = this :> IRewriteStep
+            let dir = step.AssemblyConstants.TryGetValue AssemblyConstants.OutputPath |> function
```

Comment:
I'm not convinced that the OutputPath from the compiler should be used here. Maybe by default, but probably this should be a parameter for the constructor of this Emitter.

Reply:
There needs to be a default constructor. The way to pass information is to use the AssemblyConstants dictionary (string, string dict). This is exactly how it worked before.